### ISSUE TITLE
Tank Drone overhaul

### ISSUE THREAD

**Issue by Regularitee** [CleverRaven/Cataclysm-DDA#25321]
So it has previously been discussed about how the tank drone is currently somewhat of a paradox. I'm referring to: 
- It's called a tank drone. The description uses the term "automated tank"
- It's armor is at maximum probably comparable to a humvee (~30 armor, compared to ~15 that turrets and security drones have, or ~200 that heavy power armor has). Despite it being a tank "clad in depleted uranium plating."
- Its body weighs 1.4 tons. Based on modern UGV weights, it would be smaller than a fridge. 
- This is ignoring the fact it can carries a 3-ton, 120mm cannon. Yes, a 3-ton weapon on a vehicle whose full weight is only 1.4 tons.

In other words, despite its description as being a tank, it fails every criteria of being a tank (armor, size, weight), except its main gun. Hence I am proposing we do one of these two:

1) Retcon the tank drone's description and armament to make it a smaller UGV like below. For reference, this is the 0.7 ton Gladiator ![](https://upload.wikimedia.org/wikipedia/commons/c/cb/Gladiator_240G.jpg)
This would be the safer option, as it does not introduce unprecedented level of threat that an actual tank would. And downgrade its 120mm cannon to maybe the 40mm grenade launcher.

2) Rework the vehicle's stats to make it a proper tank as lore suggests, increasing its armor and weight by a significant factor. (I mean, 30 armor is laughable compared to heavy power armor's 200 armor). This would obviously risk introducing some balancing issues, but we could drop its reaction and movement speed to compensate so it's easier to avoid.

**Comment by nexusmrsep** [comment]
I believe the only reason its so small, is that:
1. there is no support (yet) for multi tile monsters,
2. there is no support for actual autonomous vehicles.

**Comment by mvm900** [comment]
I still don't get why we're comparing things to modern tanks given that, modern tanks are usually pretty old aside from some fringe cases and this is a drone, not a tank meant to house crew. Current drones are big and bulky of course, but parts can easily be miniaturized both as technology naturally increases and with dedicated effort. 

But it's use isn't necessarily described, nor the _exact_ size of the shell (120mm only explains the diameter, not the length. And lore wise, not gameplay wise.) and not only that but being a drone you don't have to worry about anything but production costs instead of the death of a crew. So, it's entirely possible it's disposable firepower. Which would both make sense narrative-wise and allow it to maintain it's position without being too powerful in actual gameplay.

It should absolutely weigh more, but could keep the same armor and armaments for other gameplay reasons.

**Comment by Regularitee** [comment]
>I still don't get why we're comparing things to modern tanks

Because it calls itself a tank in the description? It's right there in the name, so of course people will immediately think of tanks they know about and compare it to them.

>Given that, modern tanks are usually pretty old aside from some fringe cases 

Every indication suggests this is a modern tank.
- The size classification is "huge" in-game. Or "tank-sized" on the wiki.
- The image in said wiki is a picture of an actual tank.
- The tank drone is explicitly mentioned as using the type of armor as on modern M1 tanks (depleted uranium)
- The tank drone otherwise uses existing modern weapons in the U.S. Army (M4A1 rifle, 40mm grenade launcher). Given this clear pattern of using existing U.S. Army weapons and armor, it's perfectly reasonable to assume the 120mm cannon is also the same one in use by the army (M256) unless indicated otherwise. 

>this is a drone, not a tank meant to house crew.

The majority of a tank's mass comes from its weapons, armor, and engine. It's easy to understand when considering a car as context: both have space for up to four people, have all the amenities for human habitation (temperature control, air circulation, seating, vehicle controls). Yet the average car will seat for people and weighs 1-2 tons. The average MBT crews up to four people and weighs 50-70 tons. Yet being double the width and length of a modern four-seat car, it has much less internal volume usable for each crew members. Can you say with confidence that more than a very tiny portion of the weight comes from the presence of crew?

**Comment by mvm900** [comment]
Just because it is a tank doesn't mean you should compare it to modern tanks? It's the future. Not amazingly far in the future, but far enough where one could easily say miniaturization has gotten to the point to reduce weight on similar items significantly. That was the point about comparing it to modern things. You wouldn't say any modern tank with similar weapons and armor as the Maus to have the same weight as it, would you? Or even, one merely 20-40 years after it. 

Why does size indicate it's a modern tank? The image is a reference that very clearly says 'but smaller'. Of course it'd still be big, but that's not that amazingly big. 

DU is not only not exclusive to the M1, it's not the only thing the M1 has, only later variants even get it in addition to other unknown materials. As for the gun? Well those other items are really lightweight compared to the main gun. Why reduce the weight of an M4 that weighs like 6 pounds? Worthless compared to the main gun. Not only that but you could forgo the turret entirely with an autoloader. Also peculiar to say it's a modern tank by pointing at the picture which looks nothing like any actual mass produced modern tank I've seen anywhere. It's a prototype Polish tank, though. 

Ignoring the fact that this says nowhere that it's an MBT and can just as easily be a light tank somewhat similar to the Scimitar. Incidentally, that tank pictured is actually a Polish light tank weighing merely 30,000 kg. Incredibly light compared to the M1 we just mentioned. Scimitar weighs a daintly 8 metric tons. If you look at the numbers for a large number of lightweight modern AFV's, even some with bigger guns, they can she'd tons of weight. Honestly, I believe it's more of a misinformation (Like in Fallout lore where it's said power armor can take only like 3700 joules of energy before breaking, which is merely a .308) alongside game limitations. Of course I could be incorrect but you're attempting to fix it regardless. 

Do I think the removal of crew necessitied alone could reduce a lot of weight? Absolutely. How much? I dunno. But I do know autoloaders reduce a lot of weight in a lot of different ways, depending on design. I couldn't get any real numbers for this, though, to be entirely fair. 

But based on modern light designs alone, if they were made in 2040ish where the game takes place, I think they could clock in at 20 metric or lower, being fair and all. Most (sane) militaries favor low weight, especially American, over big and heavy when possible.

**Comment by kevingranade** [comment]
I'm in favor of option one, though theres probably some kind of light antivehicular weapon you could fit on that frame, e.g. 40mm HE grenades.

**Comment by cake-pie** [comment]
I do think the tank drone would be closer in form and function to a UGV rather than a crewed tank.

However, I reject the notion that we have to make a binary choice based on existing real world UGV technology vs current tank design.

Even considering the game's current limitation of:

> there is no support (yet) for multi tile monsters

It is theoretically possible to have a pretty heavily-armored and/or -armed UGV in the 1-tile size range; we do not necessarily have to constrain ourselves to something similar to the cited example 0.7 ton Gladiator.

Even if full-blown MBT main gun (120mm L/44) might be borderline impractical, I think it is totally feasible to have a short-barreled gun such as the [M728 CEV](https://en.wikipedia.org/wiki/File:M728_Combat_Engineer_Vehicle_woodland_from_right.jpg)'s 165mm L9 -- and that's 60's technology.

---

Suggestions:
Why not both?

- rename and rework the tank drone as a near-future heavy UGV
  - size: maximum that 1-tile will accommodate, approx 1.5m dimensions
    - almost as big as [historical tankettes](https://en.wikipedia.org/wiki/Tankette)
  - ensure sane description, sane stats for weight, etc
  - short barrel main gun should be feasible
    - perhaps fairly limited ammunition

- introduce a medium UGV (closer to "cow-sized" in game parlance)
  - armaments: machine gun and/or automatic grenade launcher

---

Also recommend steering away from the "tank" terminology. There are large tanks and small tanks, so that is ambiguous. And while it is reasonable to explain to layperson that the role of a UGV is "unmanned tank", it is unhelpful that most people interpret "tank" as "main battle tank" which is in the 70+ton class. I think we're better off if we don't try to draw that kind of a comparison.

When we consider the game's monster size scale: squirrel, dog, human, cow, **TANK**  ... that last one is one helluva drastic jump. #25298 will hopefully do away with this silliness, otherwise it might be a good idea to change that to a better descriptor.

**Comment by nexusmrsep** [comment]
> though theres probably some kind of light antivehicular weapon you could fit on that frame, e.g. 40mm HE grenades.

And/or .50 cal black tipped AP rounds HMG

**Comment by Regularitee** [comment]
>short barrel main gun should be feasible
>perhaps fairly limited ammunition

If that's the case, why even use a cannon? A missile launcher would be the best bet. It has historically filled the role of the light-weight anti-tank system for most of the U.S. army (from IFVs, to humvees, to even infantry launchers). And they work well enough too; Bradley IFVs ended up collectively scoring more kills than Abrams tanks, when fighting against the Lions of Babylon (Iraqi T-72) during Gulf War 1. Since missiles release most of their propellant energy after launch, you need a far less durable (and thus very heavy) launcher to use them.

**Comment by kevingranade** [comment]
> we do not necessarily have to constrain ourselves to something similar to the cited example 0.7 ton Gladiator.

Outside of a compelling argument otherwise, I think we should.

> Even if full-blown MBT main gun (120mm L/44) might be borderline impractical, I think it is totally feasible to have a short-barreled gun such as the M728 CEV's 165mm L9 -- and that's 60's technology.

The fact that you refer to mounting a 120mm main battle tank cannon on something we are recognizing as a single-tile drone as, "borderline impractical" is deeply problematic :/  
Setting that aside, no, using a short-barreled cannon in no way makes it any better.  The recoil, the sheer mass of the mechanism, and the volume of it are completely prohibitive for anything but a full-fledged tank.  The drone we are envisioning isn't a small tank, or even just shy of a small tank.  Its smaller than even a tiny civilian vehicle.

> size: maximum that 1-tile will accommodate, almost as big as historical tankettes

I spot-checked the dimensions of these vehicles, they seem to range from 2-3 meters in length, which is a *multiple* of our tile size.  Also their main weapons were even more limited than what we were envisioning.
"They ... did not have a tank gun, instead their main weapon tended to be one or two machine guns or, rarely, a 20 mm autocannon or grenade launcher."

> short barrel main gun should be feasible

This is right out, if anything your sources argue for limiting the armaments even further.

> And/or .50 cal black tipped AP rounds HMG
Basically an anti-materiel turret with wheels/tracks.
And a rocket tube maybe?

Those are both good options, as are recoilless rifles or bazookas.

Rockets are especially good since the platform is capable of being hardened against back blast, which is one of the major shortcomings of arming infantry with them.

**Comment by mvm900** [comment]
There are... many short barreled high caliber cannons. From howitzers to simply regular tank cannons. Especially if it's a shorter round, the intent can be to put as much explosive filler into a round as possible and lob it over a relatively short distance. Pretty useful for destroying buildings, emplacements, and infantry. Granted, they're rarer now because specialization isn't really a thing most armies go for, but this can be justified by merely being an experimental variant in game. I mean, the M8 Howitzer Carrier is relatively small. And once again, if you don't need to fit humans inside, it can absolutely be smaller, without any question. Weight was thrown into question but size? There can be no argument that removal of the entire interior wouldn't allow them to be tinier. Also tankettes pretty much ended after WW2, so saying the weapons they carried then were small isn't precisely accurate as to what could be made now or even in the future. 

(Some) Vehicle mounted missile launchers tend to 'suck' (If we're gonna automate the thing) because they don't tend to reload automatically. It's either a big ass tube on a vehicle, as suggested, or it's one that folds out, or it's just a pod. Very limited ammunition both immediately available and what could be in reserve, if you can even reload the thing. Not to mention if it's intended as an anti tank weapon, then it'd be an ATGM and have very little use against any kind of 'infantry' (The zombies/you/.. really whatever else) presented in game, realistically. A recoilless rifle similarly isn't really.. easily reloadable through an automated system and I personally don't know of any that are mounted on anything more than like a jeep, due to needing a clear area behind the weapon when firing. Though, they can fire anti personell ammunition as well as anti tank. (And the bazooka is technically a recoilless rifle, but it's a rocket and of course anti tank, so..)

Also backblast isn't really a big problem on some modern missile systems, even infantry equipped ones. The javelin, for example, is specifically made to have and incredibly small backblast due to the way the missile is launched. (Firing from the tube first, then activating the actual missile engine.) 

If anything the 20mm 'restriction' placed on WW2 tankettes would work in favor of this vehicle as it would be effective versus light vehicles and infantry. Though, it could easily be upped to 30mm or the grenade launcher previously mentioned. Shit, could even possibly just give it a minigun/the famed GAU-8/some fancy future version. 

I'm a bit wary of completely stripping this thing as, at least for me, it's kind of the one threat that at least I, _personally_, have to worry about constantly in the base game, without modifying. Only enemy I really have to work around. I'd of course know this isn't the same for everyone, especially new players, but would think that at least some experienced players think the same.

**Comment by Regularitee** [comment]
So, after mulling over people's opinions so far, I'm putting forward the following proposal:

1) We remove the existing name tank drone (but keep its monster ID). This will prevent returning CDDA veterans from confusing the old tank drone with the new one. Recycling the monster ID, we create a new tank drone, with clearer references to its size (dwarf tank drone, mini-tank drone, UGV, something like that).

2) All existing weapons will be removed, except the missile launcher. It's overkill to have so many weapons on a drone, and it goes directly against its doctrinal role of a lighter, cheaper armored vehicle. Some of them also don't make sense (taser (who deploys tanks against civilians?) and flamethrowers (an obsolete weapon on a cutting edge vehicle)).

3) Drone will instead use the existing missile launcher as its main anti-armor weapon, with a limited ammo rack (or a very high reload time). This will keep the tank drone a threat against players' deathmobiles, but prevent 120mm shell spam against lone footmobile players.

4) Mid caliber weapon will be a 40mm grenade launcher. (Maybe the 30mm chain gun, but I'm worried its combination of high armor-piercing and high rate of fire may be extremely overpowered given how damage mechanics work. It would need a big accuracy nerf to be used and give the player a remote chance of survival). 

5) The low caliber weapon will be. 30 cal MG will replace the the small arms. This replaces the M4A1 assault rifle because the drone already possesses many long-range weapons, and a belt machine gun is far more appropriate for the intended role (support anti-infantry weapon). 

6) Armor values will be tripled from 30 to 90. (For reference, armored zombies have 64 armor, heavy power armor has 200 armor). The tank drone is supposed to be an end-game enemy, yet its armor can be breached by bow-and-arrows or even unarmed strikes. Additionally, such heavy armor will actually give explosives or heavy weapons clearer role in the game.

**Comment by TechyBen** [comment]
Taser was as an electrified protection on hull/armour I assume? Means melee is not an option?

Without it, the drone is more susceptible to close range, though it depends on its gun tracking. If nothing can get a jump on zero distance, then no reason to worry about threats at zero distance.

**Comment by Regularitee** [comment]
>Taser was as an electrified protection on hull/armour I assume? Means melee is not an option?
Energized armor is a bit different. It requires an actual breach of the armor to trigger (i.e., a projectile forming a connection between the inner and outer charged plates). Otherwise it would be constantly zapping random stuff and creating a huge power drain.

From an in-game perspective, the taser appears pretty worthless, between dielectric capacitors making players electricity immune and a full-auto point blank gun burst being a lot deadlier.

On an unrelated note, does anyone know what file governs monsters special attacks? I've been trying to find MULTI_ROBOT, the tank drone's special attack, but haven't had any luck.

**Comment by Regularitee** [comment]
Drone overhaul passed into mainline.
